mods/tnt/init.lua

```diff
@@ -415,6 +416,9 @@ minetest.register_node("tnt:gunpowder", {
 	on_blast = function(pos, intensity)
 		tnt.burn(pos, "tnt:gunpowder")
 	end,
+	on_burn = function(pos)
+		minetest.set_node(pos, {name = "tnt:gunpowder_burning"})
```

Comment:
I'll test this in the evening.
Honestly, to me it seems strange that when exploding the gunpowder isn't simply removed. Specially since it's likely the node it's attached to would be removed in the explosion.
I guess it's done so that the burning effects continues propagating, but I'm not sure if that really works, I'll test.